interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -129,26 +129,66 @@ namespace cling {
       return NodeContext(newCS);
     }
 
+    NodeContext VisitIfStmt(IfStmt* If) {
+      // check the condition
+      NodeContext cond = Visit(If->getCond());
+      if (!cond.isSingleStmt())
+        If->setCond((clang::Expr*)cond.getStmt());
```

Comment:
the (clang::Expr*) cast could go away if a change the return type of the method SythesizeCheck() And all the Visit methods..?